src/OpalCompiler-Core/OCLiteralList.class.st

```diff
@@ -1,46 +1,63 @@
 "
-Holds a unique ordered collection of literals
+Holds a unique ordered collection of literals.
+I uses an special dictionary inside that compares literals using #literalEquals message.
+I am optimised to guarantee that literals all unique and have an unique index.
 "
 Class {
 	#name : #OCLiteralList,
-	#superclass : #OrderedCollection,
+	#superclass : #Object,
 	#instVars : [
-		'equalitySet'
+		'literalsDict',
+		'lastIndex',
```

Comment:
It seems #lastIndex is actually a size of #literalsDict. So this variable can be replaced

Reply:
Thanks, good catch here.